### PR TITLE
Fix signer interface to conform to expected cryptoservice behavior

### LINKED ISSUE #714: Make the behavior of `GetPrivateKey` between signer and other cryptosigners
Currently, the signer returns a `nil` key, `nil` error if the `GetPrivateKey` is called on a key that doesn't exist in the trust store.

The other key stores return an error if the key doesn't exist, which means `cryptoservice` returns an error if the key doesn't exist.

`signed.Sign` assumes that if there's no error when `GetPrivateKey` on a cryptoservice, that the key isn't nil.

So if we publish without a snapshot, and the server doesn't have a snapshot key, we get a server panic (see below).  We have a test for this specific case (where no one has the snapshot key - in `client_test.TestPublishNoOneHasSnapshotKey` - but that uses an in-memory cryptoservice and not the signer.

Server panic when signer doesn't have snapshot key:
```
server_1  | {"http.request.contenttype":"multipart/form-data; boundary=e8b7767b83224d01214608e0074ea053370dfd3c1258de98889210e39bb1","http.request.host":"notary-server:4443","http.request.id":"67ab0dfc-47ca-4db3-8ec9-a9f8d905f4bb","http.request.method":"POST","http.request.remoteaddr":"172.17.0.1:50928","http.request.uri":"/v2/c8ddac9536f040adb3685a91c0836f0b/_trust/tuf/","http.request.useragent":"Go-http-client/1.1","http.response.duration":"10.503205ms","http.response.status":0,"http.response.written":0,"level":"info","msg":"response completed","time":"2016-05-03T21:38:25Z"}
server_1  | 2016/05/03 21:38:25 http: panic serving 172.17.0.1:50928: runtime error: invalid memory address or nil pointer dereference
server_1  | goroutine 388 [running]:
server_1  | net/http.(*conn).serve.func1(0xc820638380)
server_1  | 	/usr/local/go/src/net/http/server.go:1389 +0xc1
server_1  | panic(0xbb87e0, 0xc820016090)
server_1  | 	/usr/local/go/src/runtime/panic.go:426 +0x4e9
server_1  | github.com/docker/notary/tuf/signed.Sign(0x7f2019372660, 0xc82029bf50, 0xc8205f2ac0, 0xc8200e9c40, 0x1, 0x1, 0x1, 0xc8200e9c40, 0x1, 0x1, ...)
server_1  | 	/go/src/github.com/docker/notary/tuf/signed/sign.go:77 +0xa3e
server_1  | github.com/docker/notary/tuf.Repo.sign(0xc8202ed880, 0xc8205df680, 0xc820299f20, 0x0, 0x7f2019372660, 0xc82029bf50, 0xc8202f37d0, 0xcb9390, 0x4, 0x1, ...)
server_1  | 	/go/src/github.com/docker/notary/tuf/tuf.go:1062 +0x2f9
server_1  | github.com/docker/notary/tuf.(*Repo).SignSnapshot(0xc82033b770, 0xed45eabd1, 0xe11a90189, 0x12f0780, 0x12f0780, 0x0, 0x0)
server_1  | 	/go/src/github.com/docker/notary/tuf/tuf.go:1019 +0x5bb
server_1  | github.com/docker/notary/tuf.(*repoBuilder).GenerateSnapshot(0xc82033b7c0, 0xc820299f20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
server_1  | 	/go/src/github.com/docker/notary/tuf/builder.go:301 +0x560
server_1  | github.com/docker/notary/tuf.(*repoBuilderWrapper).GenerateSnapshot(0xc820017920, 0xc820299f20, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
server_1  | 	<autogenerated>:34 +0x80
server_1  | github.com/docker/notary/server/handlers.generateSnapshot(0xc8204f7449, 0x20, 0x7f2019307460, 0xc820017920, 0x7f2019372830, 0xc8202a1a20, 0x1, 0x0, 0x0)
server_1  | 	/go/src/github.com/docker/notary/server/handlers/validation.go:162 +0x3b2
server_1  | github.com/docker/notary/server/handlers.validateUpdate(0x7f2019372660, 0xc82029bf50, 0xc8204f7449, 0x20, 0xc8205df5f0, 0x1, 0x1, 0x7f2019372830, 0xc8202a1a20, 0x0, ...)
server_1  | 	/go/src/github.com/docker/notary/server/handlers/validation.go:82 +0x13f1
server_1  | github.com/docker/notary/server/handlers.atomicUpdateHandler(0x7f20193726b8, 0xc8205df200, 0x7f201937dc98, 0xc8204f7880, 0xc82068a700, 0xc8205dee70, 0x0, 0x0)
```

### PR DESCRIPTION
**Pull request by cyli**
Fixes #714 

Adds interface-specific tests so that anything that purports to implement the interface can use a standard set of tests, so hopefully all implementations behave similarly.

### CODE REVIEW

File: tuf/testutils/interfaces/cryptoservice.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	"crypto/rand"
+	"testing"
+
+	"github.com/docker/notary/trustmanager"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+	"github.com/stretchr/testify/require"
+)
+
+// These are tests that can be used to test a cryptoservice
```
Comment (riyazdf) [inline tuf/testutils/interfaces/cryptoservice.go]: these tests are awesome!  Thanks for adding interface-level tests 👍

File: tuf/testutils/interfaces/cryptoservice.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+		require.Len(t, keys, 0)
+	}
+	keys := empty.ListAllKeys()
+	require.Len(t, keys, 0)
+
+	require.NoError(t, empty.RemoveKey("nonexistent"))
+
+	require.Nil(t, empty.GetKey("nonexistent"))
+
+	k, _, err := empty.GetPrivateKey("nonexistent")
```
Comment (riyazdf) [inline tuf/testutils/interfaces/cryptoservice.go]: nit: can we test the role being empty too?

**Comment by riyazdf** [comment]
LGTM!

**Comment by endophage** [comment]
LGTM. Definitely warrants a v0.3.0-RC5

File: tuf/testutils/interfaces/cryptoservice.go (hunk 2 above)
Comment (cyli) [inline tuf/testutils/interfaces/cryptoservice.go]: Good suggestion! Will add, thanks!

**Pull request merged**